### PR TITLE
fix: search form cards structure

### PR DESCRIPTION
**Pull request by ryuk156**
Screenshot: 
image

### CODE REVIEW

**Comment by majordwarf** [comment]
I guess this PR only deals with the cards of the blog page but since we are focusing on cards in this PR, I'd like to see proper structuring. 

Example: The module page shows the following card because there is no description available. 
image

Instead of just focusing on search form cards, we should focus on all the cards (probably make the card as a component?)
Making cards a component would provide us a single source of truth to modify all the cards instead of changing code in multiple places.

**Comment by ryuk156** [comment]
> I guess this PR only deals with the cards of the blog page but since we are focusing on cards in this PR, I'd like to see proper structuring.
> 
> Example: The module page shows the following card because there is no description available.
> image
> 
> Instead of just focusing on search form cards, we should focus on all the cards (probably make the card as a component?)
> Making cards a component would provide us a single source of truth to modify all the cards instead of changing code in multiple places.

I will have a look at this and try to make it a different component

File: modules/Gooey Example/index.md [hunk 1]
```diff
@@ -7,5 +7,5 @@ category: "logic"
 tags: "Logic"
 ---
 
-![header](https://i.imgur.com/Bo7RZQv.png)
-Hello
+No information available for this module.The information will be available in upcoming updates. You can have more details about this module [here](https://github.com/Terasology)
```
Comment (jdrueckert) [inline modules/Gooey Example/index.md]:
```suggestion
Currently, there's no information available for this module. Please, refer to the method repo in [here](https://github.com/Terasology) for details.
```

File: src/components/SearchResult/SearchResult.jsx [hunk 2]
```diff
@@ -11,39 +10,21 @@ const SearchResults = ({ query, filter, results }) => (
...
-        Found 
-        {' '}
-        {results.length}
-        {' '}
-        {results.length==1 && "module matching with"} 
-        {results.length>1 && "modules matching with"} 
-        {' '}
-        "{query}"
+        Found {results.length} {results.length == 1 && "module matching with"}
+        {results.length > 1 && "modules matching with"} "{query}"
```
Comment (jdrueckert) [inline src/components/SearchResult/SearchResult.jsx:14]: if the only thing changing depending on the number of results, should we only make the "module" / "modules" part of the string depend on it and separate the "matching with" part as that stays the same?

**Review by jdrueckert — CHANGES_REQUESTED** [review]
I noted two things:
1. I don't see any preview texts: (this is the same for the module search by the way)
grafik

2. Clicking on "Next Page" will drop the search param, so if there's more search results than can be displayed on one page, you'll not be able to see the rest that's not displayed on the first page.
Related, I just noticed, that it actually showed me the 12 search results on the first page, but still showed me the "Next Page" button. It even also shows me a "Previous Page" button, if I enter something in the search field while on the second page of blog entries.
This does not happen with the modules, but I guess that's due to there not yet being so many that we need multiple pages.

**Comment by ryuk156** [comment]
> I noted two things:
> 
> 1. I don't see any preview texts: (this is the same for the module search by the way)
>    grafik
> 2. Clicking on "Next Page" will drop the search param, so if there's more search results than can be displayed on one page, you'll not be able to see the rest that's not displayed on the first page.
>    Related, I just noticed, that it actually showed me the 12 search results on the first page, but still showed me the "Next Page" button. It even also shows me a "Previous Page" button, if I enter something in the search field while on the second page of blog entries.
>    This does not happen with the modules, but I guess that's due to there not yet being so many that we need multiple pages.

For me, it is working fine for the blog and in the module, I forgot to fetch an excerpt will fix that in the next push
image


Talking about "Next" and "Previous" thing this PR only focuses on making a card as a separate component will be fixing that in the next PR

**Comment by ryuk156** [comment]
@jdrueckert Updated the PR Please have a look

File: modules/Gooey Example/index.md (hunk 1 above)
Comment (ryuk156) [reply modules/Gooey Example/index.md]: done

**Comment by jdrueckert** [comment]
Unfortunately, it's still not working for me :thinking: 
I do see the preview texts if I don't filter:
image
But as soon as I type in a letter for filtering, they disappear:
image

**Pull request closed without merging**